### ISSUE TITLE
Deal with software interruptions. 

### ISSUE THREAD

**Issue by aayushinigam** [systers/powerup-android#1103]
## Description
As a user,
my game should be paused while a software interrupt is caused,
so that I can resume playing the game without loss of any data.

Various software interrupts include - 
  1. Incoming phone call / Incoming sms
  2. Removal of battery / Device shut off
  4. Alarm
  5. Incoming alerts from other applications.
  6. Network connection loss / Network connection restoration
  7. App update reminders|
  8. Cable insertion and removal for data transfer
  9. Battery low / Battery full while charging.

### Update [Required] 
- [ ] changes in onPause() method.
- [ ] changes in onResume() method - Display of an alert dialogue asking the user to "Resume" the play.

## Definition of Done
- [ ] All of the required items are completed.
- [ ] Approval by 1 mentor.

## Estimation
2 days.

**Comment by Rimjhim28** [comment]
@aayushinigam you might also include the case when the device's screen is turned off.

**Comment by aayushinigam** [comment]
@Rimjhim28 yes right. Thanks :) 
@m-murad please approve this so that I can start working on it.

**Comment by saurabhjn76** [comment]
@aayushinigam  Yes, it is a valid issue, you can send a PR for it.

**Comment by Rimjhim28** [comment]
@aayushinigam Any updates??

**Comment by aayushinigam** [comment]
@Rimjhim28 @geekanamika I'm still working. Will soon send a PR. 
Apologies for the delay.

**Comment by geekanamika** [comment]
@aayushinigam Any updates? Discuss it on slack if you need any help :)

**Comment by aayushinigam** [comment]
@geekanamika Yes I need help. Yes I'll discuss it on slack. It was put on hold because I was busy with exams, tomorrow's the last then I'll get on with it.

**Comment by Rimjhim28** [comment]
@aayushinigam I think we can discuss this on slack now.

**Comment by pknotfound** [comment]
@Rimjhim28 @saurabhjn76 @geekanamika Is this issue still avaiable?
